numba/core/typeconv/typeconv.cpp

```diff
@@ -13,32 +13,26 @@ TCCMap::TCCMap()
 {
 }
 
-unsigned int TCCMap::hash(const TypePair &key) const {
-    const int mult = 1000003;
-    int x = 0x345678;
-    x = (x ^ key.first) * mult;
-    x = (x ^ key.second);
-    return x;
+size_t TCCMap::hash(const TypePair &key) const {
+    return std::hash<size_t>()(std::hash<Type>()(key.first)) ^
```

Comment:
Why do we hash (hash(first) ^ hash(second))? (i.e. why the "outer" hash?)

Reply:
because ^ is commutative, this is done to avoid the hash of (a,b) being the same as the hash of (b,a). Originally that was done by the mult param, but it's better just to double hash.

Reply:
Understood - many thanks!